### PR TITLE
8276105: C2: Conv(D|F)2(I|L)Nodes::Ideal should handle rounding correctly

### PR DESCRIPTION
**Pull request by shipilev**
Happens now in master:

```
$ CONF=linux-x86-server-fastdebug make run-test TEST=compiler/loopopts/superword/CoLocatePack.java TEST_VM_OPTS="-XX:UseAVX=0 -XX:UseSSE=0"
...

CompileCommand: compileonly compiler/loopopts/superword/CoLocatePack.test bool compileonly = true
 191 ConvF2L === _ 714 [[ 193 ]] !jvms: CoLocatePack::test @ bci:30 (line 70)
# To suppress the following error report, specify this argument
# after -XX: or in .hotspotrc: SuppressErrorAt=/phaseX.cpp:1128
#
# A fatal error has been detected by the Java Runtime Environment:
#
# Internal Error (/home/shade/trunks/jdk/src/hotspot/share/opto/phaseX.cpp:1128), pid=1717516, tid=1717532
# fatal error: modified node was not processed by IGVN.transform_old()
```

After JDK-8266950 (always `strictfp`), the paths in `Conv(D|F)2(I|L)Nodes::Ideal`-s start to be taken more frequently to round float/double inputs when low SSE is enabled. On those paths, we call `set_req` to rewire current node, but we still return `NULL` from `::Ideal`. I believe that is incorrect, as per `node.cpp` explanation: `NULL` indicates no graph change was done, and `this` should be returned when modification happened. So GVN predictably barfs.

Additional testing:
 - [x] Linux x86_32 `tier1` with `-XX:UseAVX=0 -XX:UseSSE=0` (now pass)
 - [x] Linux x86_32 `tier1` default (still pass)
 - [x] Linux x86_64 `tier1` default

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8276105](https://bugs.openjdk.java.net/browse/JDK-8276105): C2: Conv(D|F)2(I|L)Nodes::Ideal should handle rounding correctly

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to fb3c6a1653de30fd96fc9adc4d2525d914712a21
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to fb3c6a1653de30fd96fc9adc4d2525d914712a21

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6176/head:pull/6176` \
`$ git checkout pull/6176`

Update a local copy of the PR: \
`$ git checkout pull/6176` \
`$ git pull https://git.openjdk.java.net/jdk pull/6176/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6176`

View PR using the GUI difftool: \
`$ git pr show -t 6176`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6176.diff">https://git.openjdk.java.net/jdk/pull/6176.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Correct.

**Comment by navyxliu** [comment]
LGTM. 
Prior code style in ConvD2INode::Ideal is obscure. This patch also makes it look better.

**Comment by shipilev** [comment]
Thank you, @vnkozlov, @navyxliu. I think I need a second Hotspot (R)eviewer for this?

File: src/hotspot/share/opto/convertnode.cpp [hunk 1]
```diff
@@ -142,8 +144,10 @@ Node* ConvD2LNode::Identity(PhaseGVN* phase) {
 //------------------------------Ideal------------------------------------------
 // If converting to an int type, skip any rounding nodes
 Node *ConvD2LNode::Ideal(PhaseGVN *phase, bool can_reshape) {
-  if( in(1)->Opcode() == Op_RoundDouble )
-  set_req(1,in(1)->in(1));
+  if (in(1)->Opcode() == Op_RoundDouble) {
+    set_req(1,in(1)->in(1));
```
Comment (TobiHartmann) [inline src/hotspot/share/opto/convertnode.cpp:148]: Whitespace after `,` is missing.

**Review by TobiHartmann — APPROVED** [review]
Looks good. Is it worth adding the flag combinations to the test?

File: src/hotspot/share/opto/convertnode.cpp (hunk 1 above)
Comment (shipilev) [reply src/hotspot/share/opto/convertnode.cpp]: Fixed, thanks.

**Comment by shipilev** [comment]
> Looks good. Is it worth adding the flag combinations to the test?

Honestly, the fact that we fail on that SuperWord packing tier1 test appears to be a pure luck. Also, x86_64 would default to UseSSE >= 2 anyway, so such a config would only make sense for x86_32 running in a quite old mode. So, instead of adding the test configuration to the test, I just added some `-XX:UseSSE=0 -XX:UseAVX=0` runs to our CIs. It is unlikely we would regress this particular place again.

**Comment by shipilev** [comment]
All right, if anyone has any other comments, we can do those in follow-ups.

/integrate

**Pull request closed without merging**